0004-ledger-plugin-interface/0004-ledger-plugin-interface.md

```diff
@@ -199,82 +204,83 @@ Submit a fulfillment to a ledger. Plugin must be connected, otherwise the promis
+to you.
 
-###### Example `transfer`
+### Event: `*_prepare`
+<code style="">ledgerPlugin.on('incoming_prepare',
+  (
+    **transfer**:[Transfer](#class-transfer),
+  ) ⇒
+)</code>
 
-```js
-{
-  id: 'https//ledger.example/transfers/123',
-  account: 'https://ledger.example/accounts/connector',
-  amount: '10',
-  data: Buffer(...),
-  noteToSelf: {},
+Emitted when an outgoing/incoming transfer is prepared.
 
-  // for UTP/ATP support
-  executionCondition: 'cc:0:3:47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU:0',
+Note that the `*_prepare` event **DOES NOT** indicate that money has been transferred. The final status will only be known when either the [*_fulfill](#event-*_fulfill-) or [*_cancel](#event-*_cancel-) events are emitted.
 
-  // for UTP support
-  expiresAt: '2016-05-18T12:00:00.000Z',
-
-  // for ATP support
-  cancellationCondition: 'cc:0:3:47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU:0'
-}
-```
+The ledger plugin MUST authenticate the source for all incoming transfers, whether they include money or not.
 
-For a detailed description of these properties, please see [`IncomingTransfer`](#incomingtransfer).
+If the event is `outgoing_prepare`, then it means you prepared the transfer. `incoming_prepare` means someone prepared
+a transfer to you.
 
-#### Event: `fulfill_execution_condition`
-<code style="">ledgerPlugin.on('fulfill_execution_condition',
+### Event: `*_fulfill`
+<code style="">ledgerPlugin.on('outgoing_fulfill',
   (
     **transfer**:[Transfer](#class-transfer),
     **fulfillment**:Buffer
   ) ⇒
 )</code>
 
-Emitted when a transfer's execution condition has been fulfilled.
+Emitted when an outgoing/incoming transfer is fulfilled.
 
-If the transfer is an [`OutgoingTransfer`](#outgoingtransfer), connectors will forward the execution condition to the corresponding [`IncomingTransfer`](#incomingtransfer).
+This indicates that funds have been transfered. In order to prevent unexpected incoming funds, a ledger MAY forbid
+accounts from fulfilling a transfer who are not the transfer's receiver.
 
-#### Event: `fulfill_cancellation_condition`
-<code style="">ledgerPlugin.on('fulfill_cancellation_condition',
+If the event is `incoming_fulfill`, then it means you fulfilled the transfer. `outgoing_fulfill` means the receiver
+of your outgoing transfer has fulfilled the condition.
+
+### Event: `*_reject`
+<code style="">ledgerPlugin.on('outgoing_reject',
   (
     **transfer**:[Transfer](#class-transfer),
-    **fulfillment**:Buffer
+    **reason**:Buffer
   ) ⇒
 )</code>
 
-Emitted when a transfer's cancellation condition has been fulfilled.
+Emitted when an outgoing/incoming transfer is rejected by the receiver.
 
-If the transfer is an [`IncomingTransfer`](#incomingtransfer), connectors will forward the execution condition to the corresponding [`OutgoingTransfer`](#outgoingtransfer).
+This will cancel a transfer before the timeout. A message can be passed along with the rejection.
```

Comment:
The language here is a bit weird for an event. "This will cancel a transfer" sounds like I'm doing something, but I'm just being notified that it happened. So "This indicates that the transfer was canceled before the timeout" or something to that effect would be better.